src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/taskconfiguration/services/calendar/DueDateCalculator.java

```diff
@@ -0,0 +1,140 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.taskconfiguration.services.calendar;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.camunda.CamundaValue;
+import uk.gov.hmcts.reform.wataskmanagementapi.taskconfiguration.domain.entities.camunda.response.ConfigurationDmnEvaluationResponse;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoField;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+public class DueDateCalculator {
+
+    private static final String DUE_DATE_PREFIX = "dueDate";
```

Comment:
There are few constants duplicates in DueDateIntervalCalculator class. I think it is good to have a Calculator interface with all the common values and expose the service to the outside of this package via the interface. And make all the other services hidden inside the calendar package. calculateDueDate is the only method used outside of the calendar package. Not sure if spring allows package visibility for the componant classes though.